From b75915c70786599123ddc8989f6f96ffc9d20d47 Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Wed, 6 Nov 2013 13:22:23 -0600
Subject: [PATCH 62/71] video/da8xx-fb: allow device to go into the suspend
 state

This commit comes from branch ti-linux-3.12.y:
  git://git.ti.com/ti-linux-kernel/ti-linux-kernel.git

A call into pm_set_vt_switch() is added in this drivers probe() function
to workaround a problem where the device needs some VT interaction after
suspend is initiated for the device to continue all the way into the
suspend state.

Signed-off-by: Darren Etheridge <detheridge@ti.com>
(cherry picked from commit 8066d8452a69542706db65b079c3ba87cd801958)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/video/da8xx-fb.c |    9 +++++++++
 1 files changed, 9 insertions(+), 0 deletions(-)

diff --git a/drivers/video/da8xx-fb.c b/drivers/video/da8xx-fb.c
index 8b888f2..56fbd76 100644
--- a/drivers/video/da8xx-fb.c
+++ b/drivers/video/da8xx-fb.c
@@ -27,6 +27,7 @@
 #include <linux/platform_device.h>
 #include <linux/uaccess.h>
 #include <linux/pm_runtime.h>
+#include <linux/suspend.h>
 #include <linux/interrupt.h>
 #include <linux/wait.h>
 #include <linux/clk.h>
@@ -1537,6 +1538,14 @@ static int fb_probe(struct platform_device *device)
 	pm_runtime_enable(&device->dev);
 	pm_runtime_get_sync(&device->dev);
 
+	/*
+	 * disable creation of new console during suspend.
+	 * this works around a problem where a ctrl-c is needed
+	 * to be entered on the VT to actually get the device
+	 * to continue into the suspend state.
+	 */
+	pm_set_vt_switch(0);
+
 	/* Determine LCD IP Version */
 	switch (lcdc_read(LCD_PID_REG)) {
 	case 0x4C100102:
-- 
1.7.5.4

